### PR TITLE
Add EditorConfig files following .NET coding conventions

### PR DESCRIPTION
**Pull request by Schroedingers-Cat**
This PR adds an EditorConfig file which will support contributors in following the .NET coding conventions when using a supported text editor.

The file has been generated using VS2019 with default settings via:
Tools->Options->Text Editor->C#->Code Style->General

More info about EditorConfig: https://editorconfig.org/

A useful feature in VS is to automatically format all documents of a solution according to the rules of the EditorConfig file. However, this should only be done after the work on `feature/voice-overs` has been merged into develop to avoid diffs difficult to merge.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+#### Core EditorConfig Options ####
+
+# Indentation and spacing
+indent_size = 4
+indent_style = space
+tab_width = 4
+
+# New line preferences
+end_of_line = crlf
```
Comment (desplesda) [inline .editorconfig:15]: Can we change this to `lf`? The majority of Yarn Spinner developers work on macOS.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+dotnet_naming_style.pascal_case.required_prefix =
+dotnet_naming_style.pascal_case.required_suffix =
+dotnet_naming_style.pascal_case.word_separator =
+dotnet_naming_style.pascal_case.capitalization = pascal_case
+
+dotnet_naming_style.begins_with_i.required_prefix = I
+dotnet_naming_style.begins_with_i.required_suffix =
+dotnet_naming_style.begins_with_i.word_separator =
+dotnet_naming_style.begins_with_i.capitalization = pascal_case
```
Comment (desplesda) [inline .editorconfig:198]: No newline at end of file

Comment (Schroedingers-Cat) [reply .editorconfig:198]: Should I also change the rule in the EditorConfig to make sure that C# files have a newline at the end of file?

**Comment by Schroedingers-Cat** [comment]
Should I also change the rule in the EditorConfig to make sure that C# files have a newline at the end of file? (4cb104f008c415cec8aeff39f3739599b8d8015a)

File: .editorconfig (hunk 1 above)
Comment (Schroedingers-Cat) [reply .editorconfig]: dd32b0ee752f2df4f6e26b9a76002c19fcbaa51b

**Pull request merged**